CMakeLists.txt

```diff
@@ -152,40 +152,35 @@ if (RUN_TEST)
     PROPERTIES "IMPORTED_LOCATION"
                "${PROJECT_BINARY_DIR}/gtest/lib/libgtest_main.a"
                "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}")
-  set (GTEST_INCLUDE ${PROJECT_BINARY_DIR}/third_party/gtest/googletest/include)
-
+  set (GTEST_INCLUDE
+       "${PROJECT_BINARY_DIR}/third_party/gtest/googletest/include")
   add_library (test_util SHARED tests/test_util.cc)
   target_include_directories (test_util PRIVATE ${PROJECT_SOURCE_DIR}/include
                                                 ${GTEST_INCLUDE})
-  add_dependencies (test_util sxg libgtest libgtest_main gtest)
+  target_include_directories (test_util INTERFACE ${PROJECT_SOURCE_DIR}/include
```

Comment:
Why don't you just add PUBLIC target_include_directories instead of adding PRIVATE and INTERFACE.
https://cmake.org/cmake/help/v3.0/command/target_include_directories.html

Reply:
It's matter of preference, usually `PUBLIC` is so powerful that one should refrain from using it.
But this case, `PUBLIC` will be the best match.